src/dict.c

```diff
@@ -531,6 +531,55 @@ void *dictFetchValue(dict *d, const void *key) {
     return he ? dictGetVal(he) : NULL;
 }
 
+/* Find an element from the table, also get the plink of the entry. The entry
+ * is returned if the element is found, and the user should later call
+ * `dictFreePlinkEntry` with it in order to unlink and release it. Otherwise if
+ * the key is not found, NULL is returned.
+ *
+ * We can use like this:
+ *
+ * dictEntry *de = dictFindWithPlink(db->dict,key->ptr,&plink, &table);
+ * // Do something, but we can modify the dict
+ * dictFreePlinkEntry(db->dict,de,plink,table); // We don't need to lookup again
+ */
+dictEntry *dictFindWithPlink(dict *d, const void *key, dictEntry ***plink, int *table_index)
+{
+    dictEntry *he, *prevHe;
+    uint64_t h, idx, table;
+
+    if (dictSize(d) == 0) return NULL; /* dict is empty */
+    if (dictIsRehashing(d)) _dictRehashStep(d);
+    h = dictHashKey(d, key);
+
+    for (table = 0; table <= 1; table++) {
+        idx = h & DICTHT_SIZE_MASK(d->ht_size_exp[table]);
+        he = d->ht_table[table][idx];
+        prevHe = NULL;
```

Comment:
you can do this, and then there's no need for the `if` below:
```suggestion
        prevHe = &d->ht_table[table][idx];
        he = *prevHe;
```

Reply:
I think we still need `if`, do I miss something?

Reply:
i meant we won't need the `if (prevHe)` and it's `else`.
i.e. the `prevHe` will **always** contain the address of the pointer that leads to `he`.
in theory we can even drop the `he` variable, and just use `prevHe` or rename it to `heRef`

Reply:
If the first node, `plink` points to `prevHe`, other nodes `plink` points to `&prevHe->next`. I think we can't distinguish between them.

Reply:
i pushed a commit showing what i meant, feel free to revert if you disagree.